catroid/src/main/java/org/catrobat/catroid/formulaeditor/SensorHandler.java

```diff
@@ -397,11 +397,13 @@ public static Double getSensorValue(Sensors sensor) {
 			case DATE_YEAR:
 				return Double.valueOf(Calendar.getInstance().get(Calendar.YEAR));
 			case DATE_MONTH:
-				return Double.valueOf(Calendar.getInstance().get(Calendar.MONTH));
+				return Double.valueOf(Calendar.getInstance().get(Calendar.MONTH) + 1);
 			case DATE_DAY:
 				return Double.valueOf(Calendar.getInstance().get(Calendar.DAY_OF_MONTH));
 			case DATE_WEEKDAY:
-				return Double.valueOf(Calendar.getInstance().get(Calendar.DAY_OF_WEEK) - 1);
+				return Double.valueOf(
+						(Calendar.getInstance().get(Calendar.DAY_OF_WEEK) == Calendar.SUNDAY) ? Calendar.SATURDAY :
+								Calendar.getInstance().get(Calendar.DAY_OF_WEEK) - 1);
```

Comment:
@robertpainsi If the review should be ignored you should click the "Approve changes" button :smile: